node/src/chain/header_validation.rs

```diff
@@ -147,24 +145,21 @@ impl<'a, DB: database::DB> Validator<'a, DB> {
                     self.prev_header.seed,
                     candidate_block.height,
                 );
-                if pk != &expected_pk {
-                    anyhow::bail!("Invalid generator. Expected {expected_pk:?}, actual {pk:?}");
-                }
+
+                anyhow::ensure!(pk == &expected_pk, "Invalid generator. Expected {expected_pk:?}, actual {pk:?}");
 
                 let quorums = verify_block_cert(
                     self.prev_header.hash,
                     self.prev_header.seed,
                     self.provisioners.current(),
-                    Vote::NoCandidate,
```

Comment:
Not sure if this is done somewhere else, but we should verify the result is Fail

Reply:
you're right, added